dispatches/models/nuclear_case/chgtank.py

```diff
@@ -0,0 +1,590 @@
+For dynamic operations, multiple instances of this model indexed with time
+can be used. For this, equality constraints must be written at
+flowsheet level, i.e., 
+the final state of tank (Pres & Temp) at time (t-1) ==
+    intial state of tank (Pres & Temp) at time (t)
+
+Time window of operation
+
+This property package assumes Ideal gas behaviour of hydrogen gas.
+Adiabatic operations are assumed.
+TO DO: add constraints to enable isothermal operations
+"""
+
+# Import Pyomo libraries
+import pyomo.environ as pyo
+from pyomo.environ import (Var,
+                           Reals,
+                           NonNegativeReals,
+                           Constraint,
+                           units as pyunits)
+# from pyomo.network import Port
+from pyomo.common.config import ConfigBlock, ConfigValue, In
+
+# Import IDAES cores
+from idaes.core import (ControlVolume0DBlock,
+                        declare_process_block_class,
+                        MaterialFlowBasis,
+                        MomentumBalanceType,
+                        UnitModelBlockData,
+                        useDefault)
+from idaes.core.util.config import is_physical_parameter_block
+from idaes.core.util.constants import Constants as const
+from idaes.generic_models.properties.core.eos.ideal import Ideal
+import idaes.logger as idaeslog
+import idaes.core.util.scaling as iscale
+
+__author__ = "Naresh Susarla"
+
+# Set up logger
+_log = idaeslog.getLogger(__name__)
+
+def _make_chg_tank_config_block(config):
+    config.declare("dynamic", ConfigValue(
+        domain=In([False]),
+        default=False,
+        description="Dynamic model flag - must be False",
+        doc="""Compressed hydrogen gas tank model has not been tested 
+for dynamic operations, thus this is set to False"""))
+    config.declare("has_holdup", ConfigValue(
+            default=False,
+            domain=In([False]),
+            description="Holdup construction flag - must be False",
+            doc="""Holdup Var from the control_volume is not called.
+Instead, a Var with same name is declared in this unit model to account
+for holdup in the custom material and energy balances""",
+        ),
+    )
+    config.declare("momentum_balance_type", ConfigValue(
+        default=MomentumBalanceType.pressureTotal,
+        domain=In(MomentumBalanceType),
+        description="Momentum balance construction flag",
+        doc="""Indicates what type of momentum balance should be constructed,
+**default** - MomentumBalanceType.pressureTotal.
+**Valid values:** {
+**MomentumBalanceType.none** - exclude momentum balances,
+**MomentumBalanceType.pressureTotal** - single pressure balance for material,
+**MomentumBalanceType.pressurePhase** - pressure balances for each phase,
+**MomentumBalanceType.momentumTotal** - single momentum balance for material,
+**MomentumBalanceType.momentumPhase** - momentum balances for each phase.}"""))
+    config.declare("has_pressure_change", ConfigValue(
```

Comment:
Seeing as this can only be `True`, don't make it a config argument. Just set the value later where needed (as I see that you are doing).